ksqldb-functional-tests/src/test/resources/rest-query-validation-tests/push-queries.json

```diff
@@ -380,7 +380,7 @@
         "CREATE STREAM S1 (K INT KEY, ID bigint) WITH (kafka_topic='left_topic', value_format='JSON', WINDOW_TYPE='Tumbling', WINDOW_SIZE='2 SECONDS');",
         "CREATE STREAM S2 (K INT KEY, ID bigint) WITH (kafka_topic='right_topic', value_format='JSON', WINDOW_TYPE='Tumbling', WINDOW_SIZE='2 SECOND');",
         "CREATE STREAM OUTPUT as SELECT * FROM S1 JOIN S2 WITHIN 1 MINUTE ON S1.K = S2.K;",
-        "SELECT * FROM OUTPUT EMIT CHANGES LIMIT 2;"
+        "SELECT * FROM OUTPUT EMIT CHANGES LIMIT 3;"
```

Comment:
It is so interesting that setting a longer commit interval gives us a new result in the middle...

Reply:
Heh, yep!

Reply:
I think this is likely to be due to the same issue of https://issues.apache.org/jira/browse/KAFKA-4609

Basically, the join operator is composed of a materialized node first, then a join node, on each side. The materialized node's state store could be cached, and the cache is being flushed independently. When the flush happens, they could be triggering the join twice on each side respectively and generate duplicates.

In the past with commit interval == 0, we effectively disabled any caching effect so we did not see this.